receiver/hostmetricsreceiver/internal/scraper/nfsscraper/nfs_scraper.go

```diff
@@ -5,26 +5,45 @@ package nfsscraper // import "github.com/open-telemetry/opentelemetry-collector-
 
 import (
 	"context"
+	"time"
 
 	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
 	"go.opentelemetry.io/collector/pdata/pmetric"
 	"go.opentelemetry.io/collector/scraper"
+	"go.opentelemetry.io/collector/scraper/scrapererror"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/hostmetricsreceiver/internal/scraper/nfsscraper/internal/metadata"
 )
 
+const (
+	nfsMetricsLen  = 97
+	nfsdMetricsLen = 113
+)
+
 // nfsScraper for NFS Metrics
 type nfsScraper struct {
 	settings scraper.Settings
 	config   *Config
 	mb       *metadata.MetricsBuilder
 
-	// for mocking
+	getNfsStats  func() (*NfsStats, error)
+	getNfsdStats func() (*NfsdStats, error)
+
+	nfsStats  *NfsStats
+	nfsdStats *NfsdStats
 }
 
-// newNfsScraper creates a metric scraper for NFS metrics.
-func newNfsScraper(_ context.Context, settings scraper.Settings, cfg *Config) *nfsScraper {
-	return &nfsScraper{settings: settings, config: cfg}
+// newNfsScraper creates an NFS Scraper related metric
+func newNfsScraper(settings scraper.Settings, cfg *Config) *nfsScraper {
+	return &nfsScraper{
+		settings:     settings,
+		config:       cfg,
+		getNfsStats:  getOSNfsStats,
+		getNfsdStats: getOSNfsdStats,
+		nfsStats:     nil,
```

Comment:
These are pointer fields so they're automatically 0-initialized to nil and don't need to be explicitly set here.